app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/BluetoothViewModel.kt

```diff
@@ -220,7 +224,15 @@ class BluetoothViewModel @Inject internal constructor(
     ) {
         when (messagesAction) {
             is MessagesAction.HealthSummaryAction -> {
-                handleHealthSummaryAction(messageId)
+                handleHealthSummaryAction(messageId).onFailure {
+                    return // on failure, we don't want to dismiss the message
```

Comment:
very minor as I tend to avoid comments and want the code to speak instead 😅 . Before when I would create a `var shouldDismissMessage = isDismissable` and in both failures here force set it to false `shouldDismissMessage = false`